app/controllers/admin/facilities_controller.rb

```diff
@@ -5,44 +5,35 @@ class Admin::FacilitiesController < AdminController
 
   before_action :set_facility, only: [:show, :edit, :update, :destroy]
   before_action :set_facility_group, only: [:show, :new, :create, :edit, :update, :destroy]
+  before_action :authorize_facility, only: [:show, :new, :create, :edit, :update, :destroy]
   before_action :initialize_upload, :validate_file_type, :validate_file_size, :parse_file,
     :validate_facility_rows, if: :file_exists?, only: [:upload]
 
   def index
-    authorize([:manage, :facility, Facility])
+    authorize([:upcoming, :manage, Facility], :allowed?)
 
-    if searching?
-      facilities = policy_scope([:manage, :facility, Facility]).search_by_name(search_query)
-      facility_groups = FacilityGroup.where(facilities: facilities)
+    facilities =
```

Comment:
We were previously scoping by `Organization` here, since our original permissions system could walk the scope tree bi-directionally, this is not a thing that we would do anymore for simplicity, that means however, we need to fetch the organizations back up top from the facilities in this case, rather than the other way around.